Sources/Controllers/2.0/OADeleteCustomFiltersTableViewController.mm

```diff
@@ -0,0 +1,226 @@
+    self.tableView.estimatedRowHeight = 49.;
+
+    self.deleteButton.layer.cornerRadius = 9.0;
+}
+
+- (void)applyLocalization
+{
+    self.titleView.text = OALocalizedString(@"delete_custom_categories");
+    [self.cancelButton setTitle:OALocalizedString(@"shared_string_cancel") forState:UIControlStateNormal];
+    [self.deleteButton setTitle:OALocalizedString(@"shared_string_delete") forState:UIControlStateNormal];
+}
+
+- (IBAction)onCancelButtonClicked:(id)sender
+{
+    [self.navigationController popViewControllerAnimated:YES];
+}
+
+- (IBAction)onDeleteButtonClicked:(id)sender
+{
+    if (self.delegate)
+        [self.delegate removeFilters:_selectedItems];
+    [self.navigationController popViewControllerAnimated:YES];
+}
+
+- (void)selectDeselectGroup:(id)sender
+{
+    BOOL shouldSelect = _selectedItems.count == 0;
+    if (!shouldSelect)
+        [_selectedItems removeAllObjects];
+    else
+    {
+        [_selectedItems addObjectsFromArray:_items];
+        [_selectedItems removeObjectAtIndex: 0];
+    }
+
+    for (NSInteger i = 1; i < _items.count; i++)
+    {
+        if (shouldSelect)
+            [self.tableView selectRowAtIndexPath:[NSIndexPath indexPathForRow:i inSection:0] animated:NO scrollPosition:UITableViewScrollPositionNone];
+        else
+            [self.tableView deselectRowAtIndexPath:[NSIndexPath indexPathForRow:i inSection:0] animated:NO];
+    }
+    [self.tableView beginUpdates];
+    [self.tableView reloadRowsAtIndexPaths:@[[NSIndexPath indexPathForRow:0 inSection:0]] withRowAnimation:UITableViewRowAnimationAutomatic];
+    [self.tableView headerViewForSection:0].textLabel.text = [[NSString stringWithFormat:OALocalizedString(@"selected_of"), (int)_selectedItems.count, _items.count - 1] upperCase];
+    [self.tableView endUpdates];
+    [self.tableView reloadSections:[NSIndexSet indexSetWithIndex:0] withRowAnimation:UITableViewRowAnimationNone];
+}
+
+- (void)selectDeselectItem:(NSIndexPath *)indexPath
+{
+    if (indexPath.row > 0)
+    {
+        [self.tableView beginUpdates];
+        OAPOIUIFilter *filter = _items[indexPath.row];
+        if ([_selectedItems containsObject:filter])
+            [_selectedItems removeObject:filter];
+        else
+            [_selectedItems addObject:filter];
+        [self.tableView headerViewForSection:indexPath.section].textLabel.text = [[NSString stringWithFormat:OALocalizedString(@"selected_of"), (int) _selectedItems.count, _items.count - 1] upperCase];
+        [self.tableView endUpdates];
+        [self.tableView reloadRowsAtIndexPaths:@[[NSIndexPath indexPathForRow:0 inSection:indexPath.section], indexPath] withRowAnimation:UITableViewRowAnimationNone];
+    }
+}
+
+#pragma mark - UITableViewDataSource
+
+- (nonnull UITableViewCell *)tableView:(nonnull UITableView *)tableView cellForRowAtIndexPath:(nonnull NSIndexPath *)indexPath
+{
+    NSString* identifierCell = indexPath.row == 0 ? kCellTypeSelectionButton : kCellTypeTitle;
```

Comment:
This approach is incorrect. Instead of casting the cell, please check against the identifier and create an appropriate cell (see any other view controller with the tableview in it)